build.gradle

```diff
@@ -25,9 +26,8 @@ repositories {
 }
 
 configurations {
-    deployerJars
-    integrationTestCompile.extendsFrom testCompile
-    integrationTestRuntime.extendsFrom testRuntime
+    integrationTestImplementation.extendsFrom implementation
```

Comment:
Do we also have the dependencies of `testImplementation`?

Reply:
There are dependencies declared in `testImplementation` configuration for default `test` source set.
But they don't seem to be required for integration tests which are passing without them.
The declaration of `integrationTestTestImplementation` does not appear to be recommanded in Gradle documentation
https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests

For specific integration tests dependencies, we can directly define them in `integrationTestImplementation`
The way I understand it, it is like this:
* unit tests: `implementation` -> `testImplementation`
* integration tests: `implementation` -> `integrationTestImplementation`

It is as if you mostly should not need the same dependencies for unit tests and integration tests.

Reply:
I see your point and it makes sense yes. What I find strange, is that `JUnit` for example is declared as a `testImplementation` which means, according to the message above, that it should not be available in `integrationTestImplementation`. That's why I assumed we need to extend `testImplementation` to have dependencies of both production and test.

Reply:
I missed a point. JUnit dependencies where handled with the third extension which I didn't modify:
`integrationTestImplementation.extendsFrom testImplementation`

This means the above extensions could be simplified because `testImplementation` configuration already extends `implementation`.
A new commit is coming where the extension of `implementation` configuration has been removed.